e2e/test/scenarios/dashboard/tabs.cy.spec.js

```diff
@@ -351,6 +354,60 @@ describe("scenarios > dashboard > tabs", () => {
     cy.get("@publicFirstTabQuery").should("have.been.calledOnce");
     cy.get("@publicSecondTabQuery").should("have.been.calledOnce");
   });
+
+  it("should apply filter and show loading spinner when changing tabs (#33767)", () => {
+    visitDashboard(ORDERS_DASHBOARD_ID);
+    editDashboard();
+    createNewTab();
+    saveDashboard();
+
+    goToTab("Tab 2");
+    editDashboard();
+    openQuestionsSidebar();
+    sidebar().within(() => {
+      cy.findByText("Orders, Count").click();
+    });
+
+    cy.findByTestId("dashboard-header").within(() => {
+      cy.icon("filter").click();
+    });
+
+    popover().within(() => {
+      cy.contains("Time").click();
+      cy.findByText("Relative Date").click();
+    });
+
+    // Auto-connection happens here
+    selectDashboardFilter(getDashboardCard(0), "Created At");
+    saveDashboard();
+
+    cy.intercept(
+      "POST",
+      "/api/dashboard/*/dashcard/*/card/*/query",
+      delayResponse(500),
```

Comment:
This is needed so that the test is deterministic and won't flake because of too fast responses.